openapi.yaml

```diff
@@ -308,6 +342,270 @@ components:
         - truck
         - camper
         - trailer
+    ReferenceDataResponse:
+      description: Represents the reference data.
+      type: object
+      properties:
+        bodyTypes:
+          description: Vehicle body types.
+          type: array
+          items:
+            $ref: "#/components/schemas/BodyType"
+        conditionTypes:
+          description: Vehicle condition types.
+          type: array
+          items:
+            $ref: "#/components/schemas/ConditionType"
+        colors:
+          description: Vehicle colors.
+          type: array
+          items:
+            $ref: "#/components/schemas/Color"
+        driveTypes:
+          description: Vehicle drive types.
+          type: array
+          items:
+            $ref: "#/components/schemas/DriveType"
+        equipmentSearchAttributes:
+          description: Vehicle equipment search attributes.
+          type: array
+          items:
+            $ref: "#/components/schemas/EquipmentSearchAttribute"
+        equipmentSearchAttributeGroups:
+          description: Vehicle equipment search attribute groups.
+          type: array
+          items:
+            $ref: "#/components/schemas/EquipmentSearchAttributeGroup"
+        equipmentSearchAttributeGroupMapping:
+          type: object
+          description: Mapping between EquipmentSearchAttribute and EquipmentSearchAttributeGroup.
+          additionalProperties:
+            type: string
+            nullable: true
+          example:
```

Comment:
shouldn't we add examples for all of them or?

Reply:
it's smart enough to add the examples itself when it's only a simple structure that uses enums in this case

this is the response sample that it creates: 

```
{
  "bodyTypes": [
    "alcove"
  ],
  "conditionTypes": [
    "new"
  ],
  "colors": [
    "anthracite"
  ],
  "driveTypes": [
    "rear"
  ],
  "equipmentSearchAttributes": [
    "abs"
  ],
  "equipmentSearchAttributeGroups": [
    "air-condition"
  ],
  "equipmentSearchAttributeGroupMapping": {
    "xenon-headlights": "headlights"
  },
  "fuelTypes": [
    "cng-petrol"
  ],
  "transmissionTypes": [
    "automatic"
  ]
}
```

if you remove the example for mapping, it would be 

```
"equipmentSearchAttributeGroupMapping": {
"property1": "string",
"property2": "string"
}
```
which sucks, so i've added a manual example

Reply:
now that doesn't mean that we shouldn't add our own examples everywhere if we decide to 🤷